actors/builtin/multisig/multisig_test.go

```diff
@@ -75,10 +78,208 @@ func TestConstruction(t *testing.T) {
+
+		actor.mustContainTransactions(rt, multisig.MultiSigTransaction{
+			To:       chuck,
+			Value:    abi.NewTokenAmount(value),
+			Method:   abi.MethodNum(method),
+			Params:   nilParams,
+			Approved: []addr.Address{anne},
+		})
+	})
+
+	t.Run("propose with threshold met", func(t *testing.T) {
+		const unlockDuration = int64(0)
+		const numApprovals = int64(1)
+		const method = int64(0)
+		const value = int64(10)
+
+		var nilParams = abi.MethodParams{}
+		var signers = []addr.Address{anne, bob}
+
+		rt := builder.WithBalance(abi.NewTokenAmount(20), abi.NewTokenAmount(0)).Build()
+
+		actor.mustConstructMultisigActor(rt, numApprovals, unlockDuration, signers...)
+
+		rt.ExpectSend(chuck, abi.MethodNum(method), nilParams, abi.NewTokenAmount(value), nil, 0)
+
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.mustPropose(rt, chuck, value, method, nilParams)
+
+		actor.mustContainTransactions(rt)
+	})
+
+	t.Run("fail propose with threshold met and insufficient balance", func(t *testing.T) {
+		const unlockDuration = int64(0)
+		const numApprovals = int64(1)
+		const method = int64(0)
+		const value = int64(10)
+
+		var signers = []addr.Address{anne, bob}
+		var nilParams = abi.MethodParams{}
+
+		rt := builder.WithBalance(abi.NewTokenAmount(0), abi.NewTokenAmount(0)).Build()
+
+		actor.mustConstructMultisigActor(rt, numApprovals, unlockDuration, signers...)
+
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.mustProposeWithExitCode(rt, chuck, value, method, nilParams, exitcode.ErrInsufficientFunds)
+
+		actor.mustContainTransactions(rt, multisig.MultiSigTransaction{
+			To:       chuck,
+			Value:    abi.NewTokenAmount(value),
+			Method:   abi.MethodNum(method),
+			Params:   nilParams,
+			Approved: []addr.Address{anne},
+		})
+	})
+
+	t.Run("fail propose from non-signer", func(t *testing.T) {
+		// non-signer address
+		richard := newSecpAddr(t, 105)
+		const unlockDuration = int64(0)
+		const numApprovals = int64(2)
+		const method = int64(0)
+		const value = int64(10)
+
+		var signers = []addr.Address{anne, bob}
+		var nilParams = abi.MethodParams{}
+
+		rt := builder.Build()
+
+		actor.mustConstructMultisigActor(rt, numApprovals, unlockDuration, signers...)
+
+		rt.SetCaller(richard, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.mustProposeWithExitCode(rt, chuck, value, method, nilParams, exitcode.ErrForbidden)
+
+		actor.mustContainTransactions(rt)
+	})
+}
+
+//
+// Helper methods for calling multisig actor methods
+//
+
+type msActorHarness struct {
+	multisig.MultiSigActor
+	t testing.TB
+}
+
+func (h *msActorHarness) mustConstructMultisigActor(rt *mock.Runtime, numApprovalsThresh, unlockDuration int64, signers ...addr.Address) {
```

Comment:
More concise names possible here, like `mustConstruct()`.

Since the methods call `verify`, which has the side-effect of resetting expectations, I think these methods should be named `verifyX` to make that more clear. They don't match the usual "must" convention of panicing (nor do they "require" fail hard).

So `verifyConstruct.`